cmd/ncproxy/store.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/errdefs"
+	"github.com/pkg/errors"
+	bolt "go.etcd.io/bbolt"
+)
+
+type computeAgentStore struct {
+	DB *bolt.DB
+}
+
+func newComputeAgentStore(db *bolt.DB) *computeAgentStore {
+	return &computeAgentStore{DB: db}
+}
+
+func (n *computeAgentStore) get(ctx context.Context, key string) (result string, err error) {
+	if err := n.DB.View(func(tx *bolt.Tx) error {
+		bkt := getComputeAgentBucket(tx)
+		if bkt == nil {
+			return errors.Wrapf(errdefs.ErrNotFound, "Compute Agent bucket %v", bucketKeyComputeAgent)
+		}
+		data := bkt.Get([]byte(key))
+		if len(data) == 0 {
+			return errors.Wrapf(errdefs.ErrNotFound, "Compute Agent %v", key)
+		}
+		result = string(data)
+		return nil
+	}); err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+func (n *computeAgentStore) getKeyValueMap(ctx context.Context) (map[string]string, error) {
+	content := map[string]string{}
+	if err := n.DB.View(func(tx *bolt.Tx) error {
+		bkt := getComputeAgentBucket(tx)
+		if bkt == nil {
+			return errors.Wrapf(errdefs.ErrNotFound, "Compute Agent bucket %v", bucketKeyComputeAgent)
+		}
+		err := bkt.ForEach(func(k, v []byte) error {
+			data := bkt.Get([]byte(k))
+			content[string(k)] = string(data)
+			return nil
+		})
+		return err
+	}); err != nil {
+		return nil, err
+	}
+	return content, nil
+}
+
+func (n *computeAgentStore) update(ctx context.Context, key string, value string) error {
```

Comment:
Would also be nice to have clearer names here than `update`, `key`, and `value`. Based on the type name this isn't a generic key/value store (we're tracking compute agents), so the function signatures should indicate that as well.